From acreda1234@tiscali.co.uk  Fri May  1 13:38:46 2009
From: Bryan Stephenson <acreda1234@tiscali.co.uk>
Date: Fri, 01 May 2009 21:07:53 +0100
Subject: Staging: rt2860: fix for driver RT2860 to be able to connect to WPA2 networks
To: greg@kroah.com
Cc: linux-wireless@vger.kernel.org
Message-ID: <1241208473.7989.49.camel@Linux-CM690>




Hi

patch to change a line in the cmm_wpa.c file for the rt2860 driver so it
can connect to WPA2 networks with TKIP & AES encryption


Signed-off-by: Bryan Stephenson <acreda1234@tiscali.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2860/common/cmm_wpa.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rt2860/common/cmm_wpa.c
+++ b/drivers/staging/rt2860/common/cmm_wpa.c
@@ -625,7 +625,7 @@ VOID RTMPMakeRSNIE(
 	UCHAR		PrimaryRsnie;
 	BOOLEAN		bMixCipher = FALSE;	// indicate the pairwise and group cipher are different
 	UCHAR		p_offset;
-	WPA_MIX_PAIR_CIPHER		FlexibleCipher = MIX_CIPHER_NOTUSE;	// it provide the more flexible cipher combination in WPA-WPA2 and TKIPAES mode
+	WPA_MIX_PAIR_CIPHER		FlexibleCipher = WPA_TKIPAES_WPA2_TKIPAES;	// it provide the more flexible cipher combination in WPA-WPA2 and TKIPAES mode
 
 	rsnielen_cur_p = NULL;
 	rsnielen_ex_cur_p = NULL;
